website/src/views/modules/ModuleFinderSidebar.tsx

```diff
@@ -40,6 +48,37 @@ const EXAM_FILTER_ITEMS: FilterItem[] = [
 
 const ModuleFinderSidebar: React.FC = React.memo(() => {
   const [isMenuOpen, setMenuOpen] = useState(false);
+
+  const selectedModules = useSelector((state: StoreState) => {
+    const {
+      app: { activeSemester },
+      moduleBank: { modules },
+      timetables,
+    } = state;
+    const { timetable } = getSemesterTimetable(activeSemester, timetables);
+    const allSemesterModules = getSemesterModules(timetable, modules);
+    return allSemesterModules.map((module) => getModuleSemesterData(module, activeSemester));
```

Comment:
If you can, create `utils/null.ts` and add this util function, 

``` ts
export function isNotNull<T>(t: T | undefined | null): t is T {
  return t != null;
}
``` 

then you can filter out nullish values 


```suggestion
    return allSemesterModules.map((module) => getModuleSemesterData(module, activeSemester)).filter(isNotNull);
```